### ISSUE TITLE
Rendering issue for NativeImage with forcedTexId

### ISSUE THREAD

**Issue by sahilbajaj** [korlibs/korge#299]
Issue: First frame is rendered black when using NativeImage with forcedTexId

Usecase: We are trying to render Video (KorviVideo implementation for iOS).

The problem is that unlike SurfaceTexture (in case of Android), iOS doesn't provide a mechanism to render video frames to a texture; rather it renders into a texture backed by a texture pool. So essentially, we're creating and applying a new NativeImage instance on the node (with a forcedTexId) every frame.
The issue is the immediate next frame when we pass a new NativeImage instance is NOT rendered.

Preliminary analysis suggests that tex id getter in GlTexture generates a new texture handle whenever forcedTexId is invalid (which is the case in the first frame), until actualSyncUpload is called.

To validate the theory, in the tex getter, I simply return the forcedTexId and this WORKS.

`

                if (forcedTexId >= 0) return forcedTexId

                if (source is SyncBitmapSource) {
                    return ((source as SyncBitmapSource).gen() as NativeImage).forcedTexId
                }
`

@soywiz Please advise me on the same.
We are happy to collaborate in long term and spend engineering efforts in building this.

**Comment by soywiz** [comment]
It is hard for me to understand the problem/solution directly from text here without trying.

If you mean that the `forcedTexId` is not enough on the iOS case (maybe we can have a property with a provider so the texture can be updated?), probably we will need to update `korgw` too 

Do you have a KorVI iOS prototype implementation? If you have a prototype I can try, even if it is not perfect, we can discuss a PR at korge-next, that includes all the projects so we can change all the code in a single PR to get it working:
https://github.com/korlibs/korge-next

**Comment by sahilbajaj** [comment]
To put it simply, if we keep changing the forcedTexId on a node every frame (by using a new NativeImage instance), it never gets rendered.
One way to test this is to create a gl texture, upload image data to it and now every frame apply a **new** NativeImage instance wrapping the **SAME** texture id to this Image (View) every frame.


Our implementation is in Swift as of now.
I can try sharing with you a simple prototype as a PR.

**Comment by sahilbajaj** [comment]
`

    class RSImage(bitmap: Bitmap): Image(bitmap) {
        override fun renderInternal(ctx: RenderContext) {
            this.bitmap = RSNativeImage(100, 100, 2u, 0x0DE1).slice()
        }
    }

	class RSNativeImage(width: Int, height: Int, val name2: UInt, val target2: Int?) : NativeImage(width, height, null, true) {
		override val forcedTexId: Int get() = name2.toInt()
		override val forcedTexTarget: Int get() = if (target2 != null) target2 else GL_TEXTURE_2D
	
		override fun readPixelsUnsafe(x: Int, y: Int, width: Int, height: Int, out: RgbaArray, offset: Int) {
		}
	
		override fun writePixelsUnsafe(x: Int, y: Int, width: Int, height: Int, out: RgbaArray, offset: Int) {
		}
	}

`

**Comment by soywiz** [comment]
Yes, please. If you can provide a full executable sample reproducing the issue, your current work, and a working implementation on swift I can try to help you

**Comment by sahilbajaj** [comment]
I think there's a slight difference here between the two cases.
I haven't got a chance to look at the code but Could it be that the textures created using GlTexture APIs behave differently in this case?
